spec/classes/all_roles_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+require_relative '../support/contexts/with_mocked_nodes'
+
+def puppet_role_name_from(path)
+  path.strip.gsub('/', '::').gsub(%r{^manifests}, 'nebula').gsub(%r{\.pp}, '')
+end
+
+`find manifests/role -name '*.pp'`.each_line do |file_path|
+  role_name = puppet_role_name_from(file_path)
+  describe role_name do
+    on_supported_os.each do |os, os_facts|
+      context "on #{os}" do
+        let(:facts) { os_facts }
+
+        let(:hiera_config) do
+          if role_name.start_with? 'nebula::role::webhost::htvm'
+            'spec/fixtures/hiera/hathitrust_config.yaml'
+          elsif role_name.start_with? 'nebula::role::hathitrust'
+            'spec/fixtures/hiera/hathitrust_config.yaml'
+          elsif role_name.start_with? 'nebula::role::tools_lib'
+            'spec/fixtures/hiera/tools_lib_config.yaml'
+          elsif role_name.start_with? 'nebula::role::chipmunk'
+            'spec/fixtures/hiera/chipmunk_config.yaml'
+          elsif role_name.start_with? 'nebula::role::deb_server'
+            'spec/fixtures/hiera/deb_server_config.yaml'
+          elsif role_name.start_with? 'nebula::role::deploy_host'
+            'spec/fixtures/hiera/named_instances_config.yaml'
+          else
+            'spec/fixtures/hiera/default_config.yaml'
```

Comment:
I feel like this could be better represented, like possibly as a separate configuration file, or maybe at least as an array of duples. I'm hesitant to put it in yaml, but also it might be nice anyway to have a yaml file that shows which roles require particular hieradata to exist.

Thoughts?

Reply:
The simplest thing would probably be just stuffing everything in an 'all roles' yaml, but then we'd have to maintain that list in multiple places, and it wouldn't be very helpful to look at on its own. Maybe extract to some kind of spec helper function that loads the correct hiera for a given role? Would that be useful to use in the individual role specs as well?

Reply:
Here's what I did:

```ruby
          hiera_fixture = [
            %w[nebula::role::webhost::htvm hathitrust],
            %w[nebula::role::hathitrust hathitrust],
            #...
            %w[nebula default],
          ].select { |base, _| role_name.start_with? base }.first[1]
```

Can you think of a better ruby way than that last line to say "grab the first item in an array where X is true"?

Reply:
I believe `find` does exactly that: https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-find

Reply:
Well, you would still need to append the `[1]` to it, but you wouldn't need the `first`.